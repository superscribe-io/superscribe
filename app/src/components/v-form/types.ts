import { DeepPartial, Field } from '@directus/types';

export type FormField = DeepPartial<Field> & {
	field: string;
	name: string;
	hideLabel?: boolean;
	hideLoader?: boolean;
};
