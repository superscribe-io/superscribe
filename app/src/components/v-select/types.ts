export type Option = {
	value: string | number | null;
	icon?: string;
	text?: string;
	disabled?: boolean;
	children?: Option[];
	divider?: boolean;
	selectable?: boolean;
	hidden?: boolean;
};
