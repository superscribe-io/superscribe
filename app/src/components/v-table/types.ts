export type Alignment = 'left' | 'center' | 'right';

export type HeaderRaw = {
	text: string;
	value: string;
	description?: string | null;
	align?: Alignment;
	sortable?: boolean;
	width?: number | null;
	[key: string]: any;
};

export type Header = Required<HeaderRaw>;

export type Item = {
	[key: string]: any;
};

export type ItemSelectEvent = {
	value: boolean;
	item: Item;
};

export type Sort = {
	by: string | null;
	desc: boolean;
};
