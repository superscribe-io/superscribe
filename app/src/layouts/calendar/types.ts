export type LayoutOptions = {
	template?: string;
	startDateField?: string;
	endDateField?: string;
	firstDay?: number;
	viewInfo?: {
		type: string;
		startDateStr: string;
	};
};
