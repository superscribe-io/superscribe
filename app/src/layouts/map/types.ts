import { CameraOptions } from 'maplibre-gl';

export type LayoutQuery = {
	fields: string[];
	sort: string[];
	limit: number;
	page: number;
};

export type LayoutOptions = {
	cameraOptions?: CameraOptions & { bbox: any };
	geometryField?: string;
	autoLocationFilter?: boolean;
	clusterData?: boolean;
	animateOptions?: any;
	displayTemplate?: string;
};
