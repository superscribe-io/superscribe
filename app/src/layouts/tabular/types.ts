export type LayoutOptions = {
	widths?: {
		[field: string]: number;
	};
	align?: {
		[field: string]: 'left' | 'center' | 'right';
	};
	limit?: number;
	spacing?: 'comfortable' | 'cozy' | 'compact';
};

export type LayoutQuery = {
	fields: string[];
	sort: string[];
	page: number;
	limit: number;
};
