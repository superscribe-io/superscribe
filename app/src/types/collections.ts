import { Collection as CollectionRaw, CollectionType } from '@superscribe/types';
import { TranslateResult } from 'vue-i18n';

export interface Collection extends CollectionRaw {
	name: string | TranslateResult;
	icon: string;
	type: CollectionType;
	color?: string | null;
}
