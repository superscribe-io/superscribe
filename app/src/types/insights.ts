export type Dashboard = {
	id: string;
	name: string;
	note: string;
	icon: string;
	color: string;
	date_created: string;
	user_created: string;
};
