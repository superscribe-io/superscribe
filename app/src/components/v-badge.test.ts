import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VBadge from './v-badge.vue';
import { GlobalMountOptions } from '@vue/test-utils/dist/types';

const global: GlobalMountOptions = {
	stubs: ['v-icon'],
};

test('Mount component', () => {
	expect(VBadge).toBeTruthy();

	const wrapper = mount(VBadge, {
		slots: {
			default: 'Slot Content',
		},
		global,
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('dot and bordered prop', () => {
	const wrapper = mount(VBadge, {
		props: {
			dot: true,
			bordered: true,
		},
		global,
	});

	expect(wrapper.classes()).toContain('dot');
	expect(wrapper.classes()).toContain('bordered');
	expect(wrapper.get('span').classes()).toContain('bordered');
	expect(wrapper.get('span').classes()).toContain('dot');
});

test('icon prop', () => {
	const wrapper = mount(VBadge, {
		props: {
			icon: 'close',
		},
		global,
	});

	expect(wrapper.get('v-icon-stub').attributes().name).toBe('close');
});

test('value prop', () => {
	const wrapper = mount(VBadge, {
		props: {
			value: 'My Badge',
		},
		global,
	});

	expect(wrapper.find('span.badge span').exists()).toBeTruthy();
	expect(wrapper.find('span.badge span').text()).toBe('My Badge');
});
