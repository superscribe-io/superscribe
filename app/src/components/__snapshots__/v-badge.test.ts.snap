// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `"<div data-v-a39517a7=\\"\\" class=\\"v-badge\\"><span data-v-a39517a7=\\"\\" class=\\"badge\\"><span data-v-a39517a7=\\"\\"></span></span>Slot Content</div>"`;
